src/components/LaunchFilter/reducers/launchReducer.ts

```diff
@@ -0,0 +1,36 @@
+import { LaunchFilters } from '../constants/filerConstants';
+import { LaunchActions } from '../types/launchActionTypes';
+import type { FilterAction } from '../types/LaunchActions';
+import type { LaunchesState } from '../types/LaunchesState';
+
+const initialState: LaunchesState = {
+  [LaunchFilters.PREVIOUS]: [],
+  [LaunchFilters.CREWED]: [],
+  [LaunchFilters.CALIFORNIA]: [],
+  [LaunchFilters.FLORIDA]: [],
+  isLoading: false,
+  error: '',
+};
+
+export const launchReducer = (state = initialState, action: FilterAction) => {
+  if (!action.payload) return state;
+  const { type, payload } = action;
+  const { filterValue } = payload;
+  switch (type) {
+    case `${filterValue}${LaunchActions.LAUNCH_REQUEST}`:
+      return { ...state, isLoading: true, error: '' };
+
+    case `${filterValue}${LaunchActions.LAUNCH_SUCCESS}`: {
+      const launches = payload[filterValue as unknown as keyof typeof payload];
+      return { ...state, [filterValue]: launches, isLoading: false };
+    }
+
+    case `${filterValue}${LaunchActions.LAUNCH_ERROR}`: {
+      const error = payload['error' as keyof typeof payload];
```

Comment:
write types like this are messy. Didn't find another approach?

Reply:
I found two another:
1. my previous solution - only one action interface with  optional props;
2. use AnyAction or types which use AnyAction.

Reply:
Use AnyAction then, it would be better in this case

Reply:
done